src/renderers/dom/fiber/wrappers/ReactDOMFiberInput.js

```diff
@@ -256,10 +256,7 @@ var ReactDOMInput = {
   },
 
   restoreControlledState: function(inst) {
-    if (inst._rootNodeID) {
-      // DOM component is still mounted; update
-      ReactDOMInput.updateWrapper(inst);
-    }
+    ReactDOMInput.updateWrapper(inst);
```

Comment:
When I call into this I'll make sure to only call restoreControlledState if TreeReflection.isMounted passes.